path_finding.asciidoc

```diff
@@ -269,6 +286,55 @@ If the payment fails, then it can be re-tried via a different path by updating t
 
 We'll look at what happens if the payment is "stuck" in <<stuck_payments>>. The important detail is that a stuck payment is the worst outcome because we cannot retry with another HTLC as both (the stuck one and the retry one) might go through eventually and cause a double payment.
 
-=== Stuck payments
 
+===== First attempt (path #1)
+
+Selena attempts the first path (S->A->B->R). She constructs the onion and sends it, but receives a failure code from Bob's node. Bob reports back a +temporary channel failure+ with a +channel_update+ identifying the channel B->R as the one that can't deliver. This attempt is shown in <<path_1_fail>>:
+
+[[path_1_fail]]
+.Path 1 attempt fails
+image::images/path_1_fail.png[]
+
+===== Learning from failure
+
+From this failure code, Selena will deduce that Bob doesn't have enough liquidity to deliver the payment to Rashid on that channel. Importantly, this failure narrows the uncertainty of the liquidity of that channel! Previously, Selena's node assumed that the liquidity on Bob's side of the channel was somewhere in the range (0, 4m). Now, she can assume that the liquidity is in the range (0, 999999). Similarly, Selena can now assume that the liquidity of that channel on Rashid's side is in the range (1m, 4m), instead of (0, 4m). Selena has learned a lot from this failure.
```

Comment:
Maybe we can add an infobox and express the information gain in bits? "learnt a lot" is a bit opinionated and even though we don't open a the math in this chapter I think it is worthwhile hinting that tools to express that stuff exist. We could refer to my privios paper for readers who wish to learn more